src/main/java/com/aws/greengrass/deployment/DefaultDeploymentTask.java

```diff
@@ -111,6 +114,20 @@ public DeploymentResult call() throws InterruptedException {
                     .kv("Deployment service config", deploymentServiceConfig.toPOJO().toString())
                     .log("Starting deployment task");
 
+            // Enforce deployment sequence validation using timestamps:
+            // If we have already processed a more recent deployment for a thing group, then we will drop
+            // this deployment and only process it if it is newer (or same) deployment for that thing group.
+            if (isStaleDeployment(deploymentDocument)) {
```

Comment:
Wonder if this belongs to the Jobs/Shadow helpers before even enqueuing the deployment, there's some dedupe/replace logic and this seems like another such use case

Reply:
I talked to @devnaruka about that, I don't think they have the capability to fail/reject a deployment, all they're able to do is to drop it. In this case, we want to signal the rejection back to the cloud.

Reply:
We would need to parse the deployment configuration to get the groups. We don't parse the configuration in the jobs helper.

To keep the change footprint minimal I kept it here.